graphs/possible_bipartition.py

```diff
@@ -8,5 +8,33 @@ def possible_bipartition(dislikes):
         Time Complexity: ?
         Space Complexity: ?
     """
-    pass
+    pen_1 = []
+    pen_2 = []
 
+    for puppy in dislikes:
+        if puppy in pen_1 or puppy in pen_2:
+            continue
+        pen_1.append(puppy)
+
+        puppy_queue = []
+        for puppy_rival in puppy:
+            puppy_queue.append(dislikes[puppy_rival])
+        while puppy_queue:
+            current_puppy = puppy_queue.pop(0)
+
+            for puppy_rival in current_puppy:
+                if dislikes[puppy_rival] in pen_1:
+                    if current_puppy in pen_1:
+                        return False
+                    else:
+                        if current_puppy not in pen_2:
+                            pen_2.append(current_puppy)
+                elif dislikes[puppy_rival] in pen_2:
+                    if current_puppy in pen_2:
+                        return False
+                    else:
+                        if current_puppy not in pen_1:
+                            pen_1.append(current_puppy)
+                else:
+                    puppy_queue.append(dislikes[puppy_rival])
```

Comment:
There are cases where you are appending nodes to `puppy_queue` that might already be in the queue. Imagine a case in which the current puppy you are looking at and its first neighbor share another neighbor. That second neighbor is already in the queue, but you would add it a second time here.